### PR TITLE
DOP-3080: Resolve failing unit tests in autobuilder test suite

### PR DESCRIPTION
**Pull request by drew-beckmen**
### Ticket

[DOP-3080](https://jira.mongodb.org/browse/DOP-3080)

### Notes

[DOP-2861](https://jira.mongodb.org/browse/DOP-2861) added a basic CI workflow for the autobuilder via GitHub actions. However, CI remained useless because of pre-existing failing Jest tests. This PR fixes the 11 failing tests and removes the `continue-on-error` flag from our GitHub Actions workflow `yaml` file to ensure any code regressions are caught on future PRs.

We also had to disable coverage thresholds in `jest.config.js` to ensure Jest returns a good exit status on CI. Current coverage remains ~67%, while the thresholds were set to 90%.

See comments in the diff for details on each changes/removed test.

### CODE REVIEW

File: tests/unit/job/productionJobHandler.test.ts [hunk 1]
```diff
@@ -406,39 +373,6 @@ describe('ProductionJobHandler Tests', () => {
       jobHandlerTestHelper.job.payload.prefix
     );
     expect(jobHandlerTestHelper.jobRepo.insertPurgedUrls).toBeCalledWith(jobHandlerTestHelper.job._id, purgedUrls);
-    expect(jobHandlerTestHelper.cdnConnector.purgeAll).toHaveBeenCalledTimes(0);
```
Comment (drew-beckmen) [inline tests/unit/job/productionJobHandler.test.ts:409]: `purgeAll` from legacy Fastly CDN connector also no longer used in any code

File: tests/unit/job/productionJobHandler.test.ts [hunk 2]
```diff
@@ -406,39 +373,6 @@ describe('ProductionJobHandler Tests', () => {
...
-  test('Execute build runs successfully purges all for atlas service', async () => {
-    jobHandlerTestHelper.setStageForDeploySuccess(false);
-    jobHandlerTestHelper.config.get.calledWith('shouldPurgeAll').mockReturnValue(true);
-    jobHandlerTestHelper.config.get
-      .calledWith('cdn_creds')
-      .mockReturnValue({ 'cloud-docs-osb': { id: 'sid', key: 'token' } });
-    jobHandlerTestHelper.job.payload.repoName = 'cloud-docs-osb';
-    await jobHandlerTestHelper.jobHandler.execute();
-    // Received number of calls: 0
-    expect(jobHandlerTestHelper.cdnConnector.purgeAll).toBeCalledTimes(1);
```
Comment (drew-beckmen) [inline tests/unit/job/productionJobHandler.test.ts:441]: `purgeAll` never invoked so we wouldn't expect any received calls

**Comment by rkhullar-mongodb** [comment]
Nice to see that 158 tests passing in this PR! Just had a couple of comments around the queue url / arn.

File: config/test.json [hunk 3]
```diff
@@ -22,38 +22,10 @@
...
-    "main": {
-      "id": "FASTLY_MAIN_SERVICE_ID",
-      "key": "FASTLY_MAIN_TOKEN"
-    }
-  },
-  "repo_dir": "repos",
-  "searchIndexBucket": "SEARCH_INDEX_BUCKET",
-  "searchIndexFolder": "SEARCH_INDEX_FOLDER"
+  "jobsQueueUrl": "https://sqs.us-east-2.amazonaws.com/216656347858/autobuilder-jobs-queue-dotcomstg",
+  "jobUpdatesQueueUrl": "https://sqs.us-east-2.amazonaws.com/216656347858/autobuilder-job-updates-queue-dotcomstg"
```
[the remark this answers is not in the document]
Comment (drew-beckmen) [reply config/test.json:30]: good call!

**Pull request merged**